cvmfs/catalog_mgr_rw.h

```diff
@@ -160,8 +174,72 @@ class WritableCatalogManager : public SimpleCatalogManager {
    * Directories don't have extended attributes at this point.
    */
   XattrList empty_xattrs;
+
+  /**
+   * Balancer thresholds
+   */
+  const bool is_balanced_;
+  const unsigned max_weight_;
+  const unsigned min_weight_;
+  const unsigned balance_weight_;
 };  // class WritableCatalogManager
 
+
+template <class CatalogMgrT>
```

Comment:
CatalogBalancer is templated, so the files would be catalog_balancer.h and catalog_balancer_impl.h. Since WritableCatalogManager is not templated we could get rid of the catalog_mgr_rw_impl.h file, which is good.